apps/api/test/04-report-4.e2e-spec.ts

```diff
@@ -0,0 +1,200 @@
+import { INestApplication } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import request from 'supertest';
+
+import { AppModule } from 'src/app.module';
+import { AuthGuard } from 'src/auth/auth.guard';
+import { URLS } from './constants';
+import { canActivate } from './override-guard';
+import {
+  getApplicant,
+  getStatus,
+  reportFourNumberOfApplicants,
+  reportFourExpectedResult,
+} from './report-util';
+import { STATUS, LIC_REG_STAGE, BCCNM_LICENSE } from '@ien/common';
+import {
+  addApplicant,
+  addMilestone,
+  setApp,
+  addJob,
+  deleteApplicantStatus,
+  getHAs,
+} from './report-request-util';
+
+describe('Report 4 - Number of IEN registrants in the licensing stage', () => {
+  // Tracks applicants id's and to later delete status/milestones
+  const applicantIdsOldProcess: string[] = [];
+  const applicantIdsNewProcess: string[] = [];
+  let app: INestApplication;
+  let haPcn: string;
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    })
+      .overrideGuard(AuthGuard)
+      .useValue({ canActivate })
+      .compile();
+
+    app = moduleFixture.createNestApplication();
+    await app.init();
+
+    setApp(app);
+    const { id } = await getHAs();
```

Comment:
I think `getHAs` returns an array of authorities.

Reply:
What I meant is how object deconstruction is possible for an array. We'd better define the return type of `getHAs` function.